[package] Create API reference

SplitInfinity opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #55817 [package] Minor fixes to PackageExporter docstrings
* **#55812 [package] Create API reference**

**Summary**
This commit creates a barebones API reference doc for `torch.package`.
The content is sourced from the docstrings in the source for the
`torch.package`.

**Test Plan**
Continuous integration (specifically the docs tests).

Differential Revision: [D27726816](https://our.internmc.facebook.com/intern/diff/D27726816)

Commit:
[package] Create API reference

**Summary**
This commit creates a barebones API reference doc for `torch.package`.
The content is sourced from the docstrings in the source for the
`torch.package`.

**Test Plan**
Continuous integration (specifically the docs tests).

[ghstack-poisoned]

Commit:
Update on "[package] Create API reference"


**Summary**
This commit creates a barebones API reference doc for `torch.package`.
The content is sourced from the docstrings in the source for the
`torch.package`.

**Test Plan**
Continuous integration (specifically the docs tests).

[ghstack-poisoned]

suo commented on docs/source/package.rst:
```diff
@@ -0,0 +1,37 @@
+=============
+
+API Reference
+-------------
```
Can we add some sort of disclaimer that this is an experimental API, not publically released, etc.

suo commented on docs/source/package.rst:
```diff
@@ -0,0 +1,37 @@
+
+.. autoclass:: torch.package.PackageExporter
+
+  .. automethod:: __init__
```
Is there a way to auto include all public methods? Might cause slow drift if you have to remember to add any public method to the API docs.

Commit:
Update on "[package] Create API reference"


**Summary**
This commit creates a barebones API reference doc for `torch.package`.
The content is sourced from the docstrings in the source for the
`torch.package`.

**Test Plan**
Continuous integration (specifically the docs tests).

Differential Revision: [D27726816](https://our.internmc.facebook.com/intern/diff/D27726816)

[ghstack-poisoned]
